File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    
    num=[i for i in range(1,n+1)]
    
    s=0
    e=1
    if n==1: 
        s,e=0,0
    while s<=e:
        if sum(num[s:e+1])==n:
            answer+=1
            s+=1
            e+=1
        elif sum(num[s:e+1])<n:
            e+=1
        else:
            s+=1
        
        if e==n : 
            break
        
        
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] A=new int[n+1]; //0~n
        for(int i=1;i<A.length;i++) A[i]=i;
        int s=0;
        int e=1;
        int sum=A[s]+A[e];
        while(!(e>n)){
            if(sum<n){
                e++;
                if(e>n) break;
                sum+=A[e];
            }else if(sum>n){
                sum-=A[s];
                s++;
            }else if(sum==n){
               
                answer++;
                sum-=A[s]; s++;
                e++; if(e>n) break; 
                sum+=A[e];
                
            }
        }
        
        
        return answer;
    }
}